packages/sitecore-jss-react/src/components/VisitorIdentification.tsx

```diff
@@ -3,27 +3,37 @@ import { withSitecoreContext } from '../enhancers/withSitecoreContext';
 
 interface VIProps {
   sitecoreContext: {
+    pageEditing?: boolean;
     visitorIdentificationTimestamp?: string;
   };
 }
 
 let emittedVI = false;
-let VIComponent: React.SFC<VIProps> = ({ sitecoreContext }) => {
+const VIComponent: React.FC<VIProps> = ({ sitecoreContext }) => {
   if (
-    !emittedVI &&
-    typeof document !== 'undefined' &&
-    sitecoreContext.visitorIdentificationTimestamp
+    emittedVI ||
+    typeof document === 'undefined' ||
+    sitecoreContext.pageEditing ||
+    !sitecoreContext.visitorIdentificationTimestamp
   ) {
-    emittedVI = true;
-    const script = document.createElement('script');
-    script.src = `/layouts/system/VisitorIdentification.js`;
-    script.type = 'text/javascript';
-    document.getElementsByTagName('head')[0].appendChild(script);
-
-    return (
-      <meta name="VIcurrentDateTime" content={sitecoreContext.visitorIdentificationTimestamp} />
-    );
+    // Don't emit VI script and meta tag if we've already done so,
+    // aren't rendering client-side, are in Experience Editor, or
+    // don't have a VI timestamp.
+    return null;
   }
+  emittedVI = true;
+
+  const script = document.createElement('script');
+  script.src = `/layouts/system/VisitorIdentification.js`;
+  script.type = 'text/javascript';
+
+  const meta = document.createElement('meta');
+  meta.name = 'VIcurrentDateTime';
+  meta.content = sitecoreContext.visitorIdentificationTimestamp;
+
+  const head = document.getElementsByTagName('head')[0];
```

Comment:
@ambrauer What about `document.querySelector('head')`?

Reply:
@sc-illiakovalenko Probably safer, I've switched to use.